src/v/cluster/topic_table.cc

```diff
@@ -273,6 +273,33 @@ void incremental_update(
     }
 }
 
+template<>
+void incremental_update(
+  std::optional<model::shadow_indexing_mode>& property,
+  property_update<std::optional<model::shadow_indexing_mode>> override) {
+    switch (override.op) {
+    case incremental_update_operation::remove:
+        // remove override, fallback to default
+        if (property && override.value) {
+            property = model::drop_shadow_indexing_flag(
+              *property, *override.value);
+        } else if (!override.value.has_value()) {
+            property = std::nullopt;
+        }
+        return;
+    case incremental_update_operation::set:
+        // set new value
+        if (property && override.value) {
```

Comment:
Looks like when `property` is not set initially, we won't update it, is this correct behavior?

Reply:
fixed